src/main/java/org/assertj/core/error/BasicErrorMessageFactory.java

```diff
@@ -161,4 +162,31 @@ public String toString() {
                   CONFIGURATION_PROVIDER.representation().toStringOf(arguments));
   }
 
+  /**
+   * Return the specific name that will be used in the error message.
+   *
+   * @param actual the object that is being verified .
+   * @return an String array of length two. The first string is the object (class) name, the second string is the element name.
+   */
+  public static String[] getSpecificName(Object actual) {
+    String name = actual.getClass().getSimpleName();
+    String[] s = { "", "elements" };
+    StackTraceElement[] stElements = Thread.currentThread().getStackTrace();
+    if (stElements[5].getClassName().contains("Spliterators")) {
```

Comment:
Instead of if-elseif-else, I prefer the coding style where you return the value as soon as possible.
ex: 
```
if (actual instanceof Map) return new GroupTypeDescription("map", "map entries");
```
and yes no block needed if the if-return is formatted on the same line